pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -333,37 +333,55 @@ func (c *Controller) syncNetworkWithGNP(ctx context.Context, network *networkv1.
 	return nil
 }
 
-func (c *Controller) handleGKENetworkParamSetDelete(ctx context.Context, params *networkv1.GKENetworkParamSet) error {
+// handleGNPDelete checks to see if its safe to delete the GNP resource before calling executeGNPDelete on it
+func (c *Controller) handleGNPDelete(ctx context.Context, params *networkv1.GKENetworkParamSet) error {
 	if params.Status.NetworkName == "" {
-		removeFinalizerInPlace(params)
-		return nil
+		return c.executeGNPDelete(ctx, params, nil)
 	}
 
 	network, err := c.networkClientset.NetworkingV1().Networks().Get(ctx, params.Status.NetworkName, v1.GetOptions{})
 	if err != nil {
 		if errors.IsNotFound(err) {
-			removeFinalizerInPlace(params)
-			return nil
+			return c.executeGNPDelete(ctx, params, nil)
 		}
 		return err
 	}
 
 	networkStillRefersToGNP := network.Spec.ParametersRef != nil && strings.EqualFold(network.Spec.ParametersRef.Kind, gnpKind) && network.Spec.ParametersRef.Name == params.Name
 	if !networkStillRefersToGNP {
-		removeFinalizerInPlace(params)
-		return nil
+		return c.executeGNPDelete(ctx, params, network)
 	}
 
 	if networkStillRefersToGNP && !network.InUse() {
-		removeFinalizerInPlace(params)
-		return nil
+		return c.executeGNPDelete(ctx, params, network)
 	}
 
 	// if the network is in use, this GNP object will get reconciled again when the network's in use status changes.
 
 	return nil
 }
 
+func (c *Controller) executeGNPDelete(ctx context.Context, params *networkv1.GKENetworkParamSet, network *networkv1.Network) error {
+	removeFinalizerInPlace(params)
+
+	if network != nil {
+		newNetwork := network.DeepCopy()
+		meta.SetStatusCondition(&newNetwork.Status.Conditions, v1.Condition{
+			Type:    string(networkv1.NetworkConditionStatusParamsReady),
+			Status:  v1.ConditionFalse,
+			Reason:  string(networkv1.GNPDeleted),
+			Message: fmt.Sprintf("GKENetworkParamSet resource was deleted: %v", params.ObjectMeta.Name),
+		})
+		if !reflect.DeepEqual(network.Status.Conditions, newNetwork.Status.Conditions) {
```

Comment:
I think you can skip this check. DeepEqual is an expensive function anyways. You are updating the condition right above so the only case this is false is if the status was GNPDeleted to begin with (which is impossible).